### PR TITLE
[WIP][SPARK-16638][ML][Optimizer] fix L2 reg computation in linearRegression when standarlization is false

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

when `standardization == false`
update L2 reg computation
from
`0.5 * effectiveL2regParam * sigma( ( wi / featuresStd(i) )^2 )`
to
`0.5 * effectiveL2regParam * sigma( ( wi * featuresStd(i) )^2 )`
## How was this patch tested?

Existing test.

### CODE REVIEW

**Comment by WeichenXu123** [comment]
cc @srowen Thanks!

**Comment by srowen** [comment]
It's worth double-checking with @holdenk and @dbtsai. I think this is working as intended since `WeightedLeastSquares` does show multiplying each feature by sigma. To undo it you'd need to divide the partial gradient by its square, and divide the squared coefficient value by its square too in the loss term.

I suppose the logic is that features on a larger scale end up with small coefficients and aren't penalized much in the loss function, so multiplying them by their "scale" compensates. I think this only makes sense when fitting an intercept too, but I haven't thought this through.

**Comment by WeichenXu123** [comment]
@srowen
I re-think the code again and maybe my previous idea is wrong. The intension of author may be to use w[i] / featuresStd[i] to reduce penalty on large scale dimension (because these dimension's w[i] is dominant to training result) so that it can speed up the training. 
So I close this PR and I need more thinking. Thanks for review!
